lib/heimdall_tools/aws_config_mapper.rb

```diff
@@ -0,0 +1,321 @@
+      config_rules.each do |rule|
+        response = @client.get_compliance_details_by_config_rule(config_rule_name: rule[:config_rule_name], limit: 100)
+        rule_results = response.to_h[:evaluation_results]
+        while response.next_token
+          response = @client.get_compliance_details_by_config_rule(next_token: response.next_token, limit: 100)
+          rule_results += response.to_h[:evaluation_results]
+        end
+
+        rule[:results] = []
+        rule_results.each do |result|
+          hdf_result = {}
+          # code_desc
+          hdf_result['code_desc'] = result.dig(:evaluation_result_identifier, :evaluation_result_qualifier)&.map do |k, v|
+                                      "#{k}: #{v}"
+                                    end&.join(', ')
+          # start_time
+          hdf_result['start_time'] = if result.key?(:config_rule_invoked_time)
+                                       DateTime.parse(result[:config_rule_invoked_time].to_s).strftime('%Y-%m-%dT%H:%M:%S%:z')
+                                     end
+          # run_time
+          hdf_result['run_time'] = if result.key?(:result_recorded_time) && result.key?(:config_rule_invoked_time)
+                                     (result[:result_recorded_time] - result[:config_rule_invoked_time]).round(6)
+                                   end
+          # status
+          hdf_result['status'] = case result.dig(:compliance_type)
+                                 when 'COMPLIANT'
+                                   'passed'
+                                 when 'NON_COMPLIANT'
+                                   'failed'
+                                 else
+                                   'skipped'
+                                 end
+          hdf_result['message'] = "(#{hdf_result['code_desc']}): #{result[:annotation] || 'Rule does not pass rule compliance'}" if hdf_result['status'] == 'failed'
+          rule[:results] << hdf_result
+        end
+        next unless rule[:results].empty?
+
+        rule[:results] << case rule[:compliance]
+                          when 'NOT_APPLICABLE'
+                            {
+                              'run_time': 0,
+                              'code_desc': NOT_APPLICABLE_MSG,
+                              'skip_message': NOT_APPLICABLE_MSG,
+                              'start_time': DateTime.now.strftime('%Y-%m-%dT%H:%M:%S%:z'),
+                              'status': 'skipped'
+                            }
+                          when 'INSUFFICIENT_DATA'
+                            {
+                              'run_time': 0,
+                              'code_desc': INSUFFICIENT_DATA_MSG,
+                              'skip_message': INSUFFICIENT_DATA_MSG,
+                              'start_time': DateTime.now.strftime('%Y-%m-%dT%H:%M:%S%:z'),
+                              'status': 'skipped'
+                            }
+                          end
```

Comment:
It may be worth another opinion on this, but wouldn't `INSUFFICIENT_DATA` mean `not_reviewed` and not skipped? To me insufficient data sounds like something you would need to go validate.

Reply:
I chose `skipped` here based on the schema options here https://github.com/mitre/inspecjs/blob/master/schemas/exec-json.json#L124

Is `not_reviewed` also a legal option for the `status` field?

Reply:
I did a bit more digging, not_reviewed is not a legal option, however in Heimdall an impact of 0 && a result.status of `skipped` is considered 'Not Applicable', and an impact of greater-than 0 and a result.status of `skipped` is considered 'Not Reviewed.

Reply:
Ok, so you're saying that INSUFFICIENT_DATA rules should have their impact overridden to 0?

Reply:
Is there a sample of AWS Config data to review?

Reply:
I don't necessarily concur with setting whatever "insufficient data" means to "not applicable". Need some original data for context. Thanks!